pkg/controller/replicaset/replica_set_test.go

```diff
@@ -64,6 +65,7 @@ func testNewReplicaSetControllerFromClient(client clientset.Interface, stopCh ch
 	informers := informers.NewSharedInformerFactory(client, controller.NoResyncPeriodFunc())
 
 	ret := NewReplicaSetController(
+		klog.FromContext(context.TODO()),
```

Comment:
Please use ktesting.

Reply:
@pohly `ktesting.NewTestContext(t)` requires a testing parameter to be passed into it but in this case we don't have one, so how to deal with this?

PS: I have addressed all other comments locally, and will push once the above thing is cleared

Reply:
You can either change `testNewReplicaSetControllerFromClient` to accept a `t` parameter or add a `ctx`.

Which one is better depends on how it is used: if it's the first thing that all tests do *and* they don't need a logger or ctx themselves, then doing the `NewTestContext` call here may be most suitable. The `t` parameter may also be useful for registering cleanup functions.

But if the tests also need a ctx or logger, then it's better to initialize those in the tests and pass a ctx here.

Reply:
> if it's the first thing that all tests do and they don't need a logger or ctx themselves

the above is the case with most unit tests so changed it to `testNewReplicaSetControllerFromClient(t *testing.T,...)` and then doing this `logger, _ := ktesting.NewTestContext(t)` inside

however few challenges, I'm facing:
1. the parent func of [this call](https://github.com/kubernetes/kubernetes/blob/5303bb1832cab43bccfa41159a7d1fdcaf6ffa81/pkg/controller/replicaset/replica_set_test.go#L1483) doesn't have `t` parameter
2. the parent func of [this call](https://github.com/kubernetes/kubernetes/blob/5303bb1832cab43bccfa41159a7d1fdcaf6ffa81/pkg/controller/replicaset/replica_set_test.go#L449) has benchmark `b` as input which can't be passed 

que: `The t parameter may also be useful for registering cleanup functions.` is cleanup automatic or do we need to add something for this too?

Reply:
- change `setupManagerWithGCEnabled` to take a context and use that to pass down logger and cancelation (`stopCh` -> `context.Done()`, with WithCancel in the parent)
- to deal with functions which get used in both tests and benchmarks, you can accept `tb testing.TB`

Reply:
- > change setupManagerWithGCEnabled to take a context

I changed it to take `t` from tests